universal-application-tool-0.0.1/app/services/program/Predicate.java

```diff
@@ -0,0 +1,16 @@
+package services.program;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+
+/** String representation of a predicate with variables from the core data model. */
+@AutoValue
+abstract class Predicate {
+  static Predicate create(String function, ImmutableSet<String> variables) {
+    return new AutoValue_Predicate(function, variables);
+  }
+
+  abstract String function();
+
+  abstract ImmutableSet<String> variables();
```

Comment:
Question: what's an example of a variable and how it would be used in a predicate?

Reply:
We'd like to have an easy way to get all the variables (JSON paths) referenced by a predicate, since it's not easy to parse from the expression string

Reply:
This behavior isn't needed for the milestone so I'm gonna cut it out.